From 6e2809ca732d8bb4d9bc7e2efb3f43c65edf7407 Mon Sep 17 00:00:00 2001
From: jrior001 <jriordan001@gmail.com>
Date: Wed, 15 Aug 2018 07:12:25 +0000
Subject: [PATCH 74/74] libfm_jni: Resolve unused and uninitialized variable
 errors

 * Required for devices that set QCOM_NO_FM_FIRMWARE to true.

Change-Id: I8facc8e80599d25acd8dd78fb9b03628b2fe59e5
---
 libfm_jni/Android.mk            | 1 +
 libfm_jni/FmIoctlsInterface.cpp | 7 +++++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/libfm_jni/Android.mk b/libfm_jni/Android.mk
index 5ac0135..c4579c2 100644
--- a/libfm_jni/Android.mk
+++ b/libfm_jni/Android.mk
@@ -33,6 +33,7 @@ endif
 ifeq ($(TARGET_QCOM_NO_FM_FIRMWARE),true)
     LOCAL_CFLAGS += -DQCOM_NO_FM_FIRMWARE
 endif
+LOCAL_CFLAGS += -Wno-unused-parameter
 
 LOCAL_MODULE := libfmjni
 include $(BUILD_SHARED_LIBRARY)
diff --git a/libfm_jni/FmIoctlsInterface.cpp b/libfm_jni/FmIoctlsInterface.cpp
index 90d62e1..8f8e212 100644
--- a/libfm_jni/FmIoctlsInterface.cpp
+++ b/libfm_jni/FmIoctlsInterface.cpp
@@ -48,7 +48,9 @@ int FmIoctlsInterface :: start_fm_patch_dl
     int ret;
     int init_success = 0;
     char versionStr[MAX_VER_STR_LEN] = {'\0'};
+#ifndef QCOM_NO_FM_FIRMWARE
     char prop_value[PROPERTY_VALUE_MAX] = {'\0'};
+#endif
     struct v4l2_capability cap;
 
     ALOGI("%s: start_fm_patch_dl = %d\n", __func__, fd);
@@ -108,9 +110,9 @@ int  FmIoctlsInterface :: close_fm_patch_dl
     void
 )
 {
+#ifndef QCOM_NO_FM_FIRMWARE
     int ret;
 
-#ifndef QCOM_NO_FM_FIRMWARE
     ret = property_set(SCRIPT_STOP_PROP, SOC_PATCH_DL_SCRPT);
     if(ret != PROP_SET_SUCC) {
         return FM_FAILURE;
@@ -184,13 +186,14 @@ int  FmIoctlsInterface :: set_calibration
     UINT fd
 )
 {
+
+#ifndef QCOM_NO_FM_FIRMWARE
     int ret;
     FILE *cal_fp;
     struct v4l2_ext_control ext_ctl;
     struct v4l2_ext_controls v4l2_ctls;
     char cal_data[CAL_DATA_SIZE] = {0};
 
-#ifndef QCOM_NO_FM_FIRMWARE
     cal_fp = fopen(CALIB_DATA_NAME, "r");
     if(cal_fp != NULL) {
        if(fread(&cal_data[0], 1, CAL_DATA_SIZE, cal_fp)
-- 
2.17.1

